### ISSUE TITLE
Hot Reload does not work with Dumber and Explicit Routing

### ISSUE THREAD

**Issue by nenadvicentic** [aurelia/aurelia#1485]
# 🐛 Bug Report

When using Aurelia 2 bare-minimum template, with Dumber, TypeScript and SCSS, in combination with explicit routing and a child component, *ANY* change to the content, no matter how trivial (e.g. `public message = 'Hello World!!!';`) breaks hot-reload (`npm start`). Reloaded main page looses child component content.

## 🤔 Expected Behavior

When using explicit `Routing` and a trivial child component:

*my-app.ts*:

```typescript
import { IRoute } from "@aurelia/router";

export class MyApp {
  public message = 'Hello World!!!';

  static routes: IRoute[] = [
    {
      path: '',
      component: import('./components/home-page'),
      title: 'Home',
    }
  ];
}
```
*my-app.html*:
```html
<div class="message">${message}</div>
<au-viewport></au-viewport>
```
And a child component in `./components` child folder:

*home-page.html*:

```html
<p>
    This is a paragraph that will be shown beneath the main title,
    when on the home page.
</p>
```

*home-page.ts*:

```typescript
export class HomePage {
}
```

After running `npm start`, main page, correctly, looks like:

<img src="https://user-images.githubusercontent.com/4405210/181907950-4cacba0f-110c-4040-a941-00f65da1661a.png" width="60%" />

Modifications of the source code should reflect when page reloads.

## 😯 Current Behavior

Instead, currently, any change of the code/content makes reloaded page to loose child component content.

For example, if we make a change in `my-app.ts`: `public message = 'Hello World!!';`

Automatically reloaded page will look like this:

<img src="https://user-images.githubusercontent.com/4405210/181908113-68633c0d-4171-44fc-955a-e5eee92827c6.png" width="40%" />

Any other change will do the same (e.g. modifying paragraph text content of the `home-page`).

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Aurelia          | 2.0.0-alpha.36
| Language         | Typescript
| Browser          | Chrome 103
| Bundler          | Dumber
| Operating System | Windows 11
| NPM/Node/Yarn    |  8.11.0/v16.16.0/1.22.15

**Comment by bigopon** [comment]
@3cp is this a no-support for dumber? With hmr, there will be `module.hot...` or `import.meta.hot...` etc, this will break dumber right?

**Comment by 3cp** [comment]
Dumber has no hmr, it's merely a bundler (merge modules into one or few bundle files).
Dumber has no dev server, it is a standard nodejs http server (connect + few middlewares) in dumber setup.

I thought about adding hmr to dumber (as hmr should be natural for AMD than other types of bundlers), but never got enough desire to do it :-)

Our hmr is behind an option to our plugin. It's off for our gulp-plugin, so it would not run for dumber.

**Comment by 3cp** [comment]
I don't read this issue is a hmr issue. Reload is not hmr, this is probably some bug.

**Comment by 3cp** [comment]
I will try to investigate latter today.

**Comment by 3cp** [comment]
I don't have the issue with au2 latest 2.0.0-beta.17.

Your 2.0.0-alpha.36 is quite outdated. Also your nodejs v16 is EOL.

Try latest au2 pls.

BTW, with dumber, there is no hot reload, it's a full page reload (which user loses the ui state like input and selection).

**Comment by 3cp** [comment]
Just noticed this is old issue. I assume it's a router bug that got fixed.

**Comment by nenadvicentic** [comment]
@3cp Yes, it has been a while since I created this issue.

So, I tried to recreate this issue in a new project created using `npx makes aurelia`. It seems to work OK now.

Only had to remember to register `RouterConfiguration` in `main.ts` file:

```
import Aurelia from 'aurelia';
import { RouterConfiguration } from '@aurelia/router'; 
import { MyApp } from './my-app';

Aurelia
  .register(RouterConfiguration)
  .app(MyApp)
  .start();
```

I don't think this was a bug in the router, because with Webpack everything worked fine. But I wanted to avoid Webpack. At the time Parcel had other issue #1408, so did not work properly either.
